### PR TITLE
Persist Proposal, Transaction, Commit and Event Objects (Node JS)

### PR DESCRIPTION
**Pull request by sapthasurendran**
Contributes to  #394 

Signed-off-by: sapthasurendran <saptha.surendran@ibm.com>

### CODE REVIEW

**Comment by denyeart** [comment]
@sapthasurendran If you add text like "Resolves #394" or "Closes #394" in the PR and commit message it will automatically get associated with the GitHub issue.

File: node/src/chaincodeeventsrequest.ts [hunk 1]
```diff
@@ -36,22 +36,21 @@ export interface ChaincodeEventsRequest extends Signable {
     getEvents(options?: CallOptions): Promise<CloseableAsyncIterable<ChaincodeEvent>>;
 }
 
-export interface ChaincodeEventsRequestOptions {
+export interface SignedChaincodeEventsRequestOptions {
```
Comment (bestbeforetoday) [inline node/src/chaincodeeventsrequest.ts:39]: These are still the options used when creating a ChaincodeEventsRequest so the name should remain the same.

File: node/src/gateway.ts [hunk 2]
```diff
@@ -156,6 +156,13 @@ export interface Gateway {
      */
     newSignedProposal(bytes: Uint8Array, signature: Uint8Array): Proposal;
 
+    /**
+     * Create a proposal from a serialized proposal.
+     * @param bytes - Serialized proposal.
+     * @returns A proposal.
```
Comment (bestbeforetoday) [inline node/src/gateway.ts:162]: I wonder if we would be better to have a simple, consistent description for all of these `newType(bytes: Uint8Array)` methods. How about, "Recreate a _type_ from serialized data"?
```suggestion
     * Recreate a proposal from serialized data.
     * @param bytes - Serialized proposal.
     * @returns A proposal.
```

**Review by bestbeforetoday — CHANGES_REQUESTED** [review]
One code naming change that needs to be reverted and doc wording suggestion, otherwise looks good

File: node/src/chaincodeeventsrequest.ts (hunk 1 above)
Comment (sapthasurendran) [reply node/src/chaincodeeventsrequest.ts:39]: It is already `SignedChaincodeEventsRequestOptions`

**Review by bestbeforetoday — APPROVED** [review]

**Pull request merged**